oauthproxy.go

```diff
@@ -130,7 +130,20 @@ func (u *UpstreamProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // NewReverseProxy creates a new reverse proxy for proxying requests to upstream
 // servers
 func NewReverseProxy(target *url.URL, opts *Options) (proxy *httputil.ReverseProxy) {
+
 	proxy = httputil.NewSingleHostReverseProxy(target)
+	proxy.Transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   time.Duration(opts.ProxyTimeOut) * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```

Comment:
Out of interest, where did all of these defaults come from? Maybe we could add a comment so that others can see where they came from in the future?

Reply:
this comes from the `net/http` pkg 

I have directly used `DefaultTransport` while keeping all its other field as default and timeout field as our custom one

You are right, I will add the comment